src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -170,11 +195,61 @@ private string CreateTokenPrivate(JObject payload, SigningCredentials signingCre
             var rawSignature = JwtTokenUtilities.CreateEncodedSignature(message, signingCredentials);
 
             if (encryptingCredentials != null)
-                return EncryptToken(message + "." + rawSignature, encryptingCredentials);
+                return EncryptToken(message + "." + rawSignature, encryptingCredentials, compressionAlgorithm);
             else
                 return message + "." + rawSignature;
         }
 
+        /// <summary>
+        /// Compress a JWT token string.
+        /// </summary>
+        /// <param name="token"></param>
+        /// <param name="compressionAlgorithm"></param>
+        /// <exception cref="ArgumentNullException">if 'token' is null.</exception>
+        /// <exception cref="ArgumentNullException">if 'compressionAlgorithm' is null.</exception>
+        /// <exception cref="NotSupportedException">if the compression algorithm is not supported.</exception>
+        /// <returns>Compressed JWT token bytes.</returns>
+        protected byte[] CompressToken(string token, string compressionAlgorithm)
+        {
+            if (token == null)
+                throw LogHelper.LogArgumentNullException(nameof(token));
+
+            if (string.IsNullOrEmpty(compressionAlgorithm))
+                throw LogHelper.LogArgumentNullException(nameof(compressionAlgorithm));
+
+            if (!CompressionProviderFactory.Default.IsSupportedAlgorithm(compressionAlgorithm))
+                throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(TokenLogMessages.IDX10682, compressionAlgorithm)));
+
+            var compressionProvider = CompressionProviderFactory.Default.CreateCompressionProvider(compressionAlgorithm);
+        
+            return compressionProvider.Compress(Encoding.UTF8.GetBytes(token)) ?? throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(TokenLogMessages.IDX10680, compressionAlgorithm)));
+        }
+
+        /// <summary>
+        /// Decompress JWT token bytes.
+        /// </summary>
+        /// <param name="tokenBytes"></param>
+        /// <param name="decompressionAlgorithm"></param>
+        /// <exception cref="ArgumentNullException">if 'tokenBytes' is null.</exception>
+        /// <exception cref="ArgumentNullException">if 'decompressionAlgorithm' is null.</exception>
+        /// <exception cref="NotSupportedException">if the decompression algorithm is not supported.</exception>
+        /// <returns>Decompressed JWT token</returns>
+        protected string DecompressToken(byte[] tokenBytes, string decompressionAlgorithm)
```

Comment:
Would it make sense to have this return byte[], similar to the provider

Reply:
I've changed the methods to be private.